src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/template.json

```diff
@@ -54,7 +54,17 @@
             "sku": {
                 "name": "[parameters('storageAccountType')]"
             },
-            "kind": "Storage"
+            "kind": "Storage",
+            "properties": {
+                "encryption": {
+                    "services": {
+                        "blob": {
+                            "enabled": true
```

Comment:
This isn't so much "enabled by default" as it is forced enabled. Did you want the ability to create the storage account without encryption as a non-default behavior?